experiments/book-surprisal/ocr_surprisal.py

```
@@ -0,0 +1,95 @@
+        if h is not None: rnd_hits.append(h)
+    # the fairer null: real words occurring exactly once in the whole refined
+    # book, legitimate but rare, up to three occurrences a page
+    rare=[m.span() for m in re.finditer(r"[A-Za-z]{4,}", fp["text"]) if freq.get(m.group(0))==1][:3]
+    for sp in rare:
+        h=span_hit(sp, fp["offsets"], fp["surprisal"], pct)
+        if h is not None: rare_hits.append(h)
+# **An empty control refuses rather than reporting a rate of zero**: the
+# comparison is the point of the run, and a null written as 0.0 would read
+# as a measurement.
+for name, hits in (("artifact", art_hits), ("random", rnd_hits), ("rare real word", rare_hits)):
+    if not hits: raise SystemExit(f"the {name} control scored no occurrence: nothing to compare")
```

Ruff (0.16.3)

[error] Multiple statements on one line (colon)

(E701)